core/Test/Tasty/Core.hs

```diff
@@ -259,7 +273,7 @@ data DependencyType
 data TestTree
   = forall t . IsTest t => SingleTest TestName t
     -- ^ A single test of some particular type
-  | TestGroup TestName [TestTree]
+  | TestGroup ExecutionMode TestName [TestTree]
```

Comment:
Can we pass `ExecutionMode` as an `Option` rather than a dedicated field of `TestGroup`? This would avoid any breakage to `data TestTree` and is more `tasty`, because pretty much every other execution option is passed like this.

Reply:
Clever, I like it. Will do.

Reply:
Options propagate all the way down, but there's no reason execution mode information should. The only way around that seems that `testGroupSequential` would do something like this pseudocode:

```
TestGroup [t1, t2]

=>

setSequential (TestGroup "x" [setParallel t1, setParallel t2])
```

That would ensure the options of test group `x` would evaluate to "test sequentially" while evaluation the options for t1/t2 would yield "test in parallel".

This seems kinda weird/magical, do we want to pursue this?

Reply:
I think that's fine, yes.

Reply:
Ok, done! I'll squash the new commit into its parent if we intend to merge.